playerbot/BroadcastHelper.cpp

```diff
@@ -0,0 +1,922 @@
+}
+
+bool BroadcastHelper::BroadcastSuggestSomething(
+    PlayerbotAI* ai,
+    Player* bot
+)
+{
+    BroadcastTest(ai, bot);
+
+    if (urand(1, sPlayerbotAIConfig.broadcastChanceMaxValue) <= sPlayerbotAIConfig.broadcastChanceSuggestSomething)
+    {
+        std::map<std::string, std::string> placeholders;
+        placeholders["%my_role"] = ai->GetChatHelper()->formatClass(bot, AiFactory::GetPlayerSpecTab(bot));
+
+        AreaTableEntry const* current_area = ai->GetCurrentArea();
+        AreaTableEntry const* current_zone = ai->GetCurrentZone();
+        placeholders["%area_name"] = current_area ? current_area->area_name[BroadcastHelper::GetLocale()] : BOT_TEXT("string_unknown_area");
+        placeholders["%zone_name"] = current_zone ? current_zone->area_name[BroadcastHelper::GetLocale()] : BOT_TEXT("string_unknown_area");
+        placeholders["%my_class"] = ai->GetChatHelper()->formatClass(bot->getClass());
+        placeholders["%my_race"] = ai->GetChatHelper()->formatRace(bot->getRace());
+        placeholders["%my_level"] = std::to_string(bot->GetLevel());
+
+        return BroadcastToChannelWithGlobalChance(
+            ai,
+            BOT_TEXT2("suggest_something", placeholders),
+            { {TO_GUILD, 10}, {TO_WORLD, 70}, {TO_GENERAL, 100} }
+        );
+    }
+
+    return false;
+}
+
+bool BroadcastHelper::BroadcastSuggestSomethingFunny(
+    PlayerbotAI* ai,
+    Player* bot
+)
+{
+    BroadcastTest(ai, bot);
+
+    if (urand(1, sPlayerbotAIConfig.broadcastChanceMaxValue) <= sPlayerbotAIConfig.broadcastChanceSuggestSomethingFunny)
+    {
+        //items
+        std::vector<Item*> botItems = ai->GetInventoryAndEquipItems();
+
+        std::map<std::string, std::string> placeholders;
+
+        placeholders["%random_inventory_item_link"] = botItems.size() > 0 ? ai->GetChatHelper()->formatItem(botItems[rand() % botItems.size()]) : BOT_TEXT("string_empty_link");
+
+        placeholders["%my_role"] = ai->GetChatHelper()->formatClass(bot, AiFactory::GetPlayerSpecTab(bot));
+        AreaTableEntry const* current_area = ai->GetCurrentArea();
+        AreaTableEntry const* current_zone = ai->GetCurrentZone();
+        placeholders["%area_name"] = current_area ? current_area->area_name[BroadcastHelper::GetLocale()] : BOT_TEXT("string_unknown_area");
+        placeholders["%zone_name"] = current_zone ? current_zone->area_name[BroadcastHelper::GetLocale()] : BOT_TEXT("string_unknown_area");
+        placeholders["%my_class"] = ai->GetChatHelper()->formatClass(bot->getClass());
+        placeholders["%my_race"] = ai->GetChatHelper()->formatRace(bot->getRace());
+        placeholders["%my_level"] = std::to_string(bot->GetLevel());
+
+        return BroadcastToChannelWithGlobalChance(
+            ai,
+            BOT_TEXT2("suggest_something_funny", placeholders),
+            { {TO_GUILD, 10}, {TO_WORLD, 70}, {TO_GENERAL, 100} }
+        );
+    }
+
+    return false;
+}
+
+bool BroadcastHelper::BroadcastSuggestAnal(
```

Comment:
that's literally just for "anal [Item link]", I wanted to separate it, so it corresponds to the context

Reply:
Is that short for analyze?

Reply:
no, that's literally anal

Reply:
OK, since I can't find any reference to this anywhere does that mean your bots randomly suggest shoving an item or quest up someone's ass? The fuck?

What kind of useless feature is that

Reply:
no, they will randomly suggest "anal [Item link", and I specifically separated it so it could be disabled easily or have a very low chance

Reply:
I can't wait for random dads who play on CMaNGOS with their kids yelling at us on our discord because they had to explain to their 8 year old what "anal sex" is 🙄

Reply:
will set default to 0 then, and everyone will be happy

Reply:
made it be disabled by default

Reply:
(fixed)

Reply:
ty, I am a dat, and my dauther is 9
:)